app/views.py

```diff
@@ -22,9 +23,34 @@
 from boto.s3.connection import S3Connection
 
 import mimetypes
+import functools
 
 views = Blueprint('views', __name__)  # pylint: disable=invalid-name
 
+def full_registration_required(func):
+    '''
+    A view decorator that requires both login *and* full completion
+    of the registration process. If the user isn't logged in, they
+    are redirected to login; if they are logged in but not fully
+    registered, they are redirected to complete the registration
+    process.
+
+    Note that if the user is not marked as fully registered but
+    nonetheless meets the criteria for full registration, they will
+    be marked as fully registered and granted access to the view.
+    '''
+
+    @functools.wraps(func)
+    @login_required
+    def decorated_view(*args, **kwargs):
+        if not current_user.has_fully_registered:
+            if current_user.matches_criteria_for_full_registration():
+                current_user.set_fully_registered()
```

Comment:
Hm.  To reply:


> 1. They could be migrated from a version of a deployment that didn't have registration in-place (though I suppose we could alternatively create a data migration for this).

I think we should handle this explicitly in the migration.

> 2. They could answer questions at /my-expertise instead of registration step 3 (alternatively, we could require full registration to access that endpoint).

I think we should require full registration for them to access that endpoint, especially if that buys us code simplicity.

> 3. There could just be weird edge cases in the future where a user should be considered as fully registered but doesn't have the associated event; we don't want to inconvenience users if this happens by needlessly redirecting them to the registration flow.

While this could be true, it's not yet.  I don't think we need to worry about it.

In short, I think we should equate `login_required` with `full_registration_required`, especially if that buys us code simplicity.